### ISSUE TITLE
REPL tab completion: some completions are missing

### ISSUE THREAD

**Issue by yha** [JuliaLang/julia#40356]
After `using Flux`,
```
julia> Flux.CU<tab>
```
completes to `Flux.CUDAint`, skipping `Flux.CUDA`.
Also, `Flux.CUDA.all<tab>` completes to `Flux.CUDA.alloc`, skipping `Flux.CUDA.allowscalar`. An additional tab shows
```
julia> Flux.CUDA.alloc
alloc          alloc_stats     alloc_timings   alloc_to        allocated       allocated_lock
```

```
julia> versioninfo()
Julia Version 1.6.0
Commit f9720dc2eb (2021-03-24 12:55 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-9800X CPU @ 3.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake-avx512)
```
`Flux` version is 0.12.1

**Comment by regier21** [comment]
On your second example, ```allowscalar``` does not begin with ```alloc```. Was this a typo on your end or am I missing something?

**Comment by yha** [comment]
That was a mistake. Typing `Flux.CUDA.all<tab>` completes to `alloc` and an additional tab displays the list above. I will edit the original.

**Comment by Liozou** [comment]
I think this may be a duplicate of #36529 (the case of REPL completion is noted in https://github.com/JuliaLang/julia/issues/36529#issuecomment-673493581). This comes from the fact that CUDA is a separate package imported by Flux.
Since the example is a bit different (CUDA is not defined within Flux), it may be good to leave this open while the other issue is resolved; you can also check whether PR #42094 solves it.

**Comment by PetrKryslUCSD** [comment]
Is this the same thing as reported here https://discourse.julialang.org/t/where-is-sparse-ldlt/78929/2?
SparseArrays.cho does not complete to SparseArrays.cholesky.
